includes/Core/Modules/Modules.php

```diff
@@ -566,10 +566,13 @@ protected function get_registry() {
 	 * @return Module_Registry
 	 */
 	protected function setup_registry() {
-		$registry = new Module_Registry();
+		$registry          = new Module_Registry();
+		$available_modules = (array) apply_filters( 'googlesitekit_available_modules', array_keys( $this->core_modules ) );
 
-		foreach ( $this->core_modules as $core_module ) {
-			$registry->register( $core_module );
+		foreach ( $this->core_modules as $slug => $module ) {
+			if ( in_array( $slug, $available_modules, true ) ) {
```

Comment:
Hey @eugene-manuilov when looking at this based on the current IB at `setup_registry` the modules were not yet set so this filter should run instead at `get_available_modules()` method or at `get_active_modules()` as at this point all the modules were created and we know to have access to the property `force_active` on each module.

Let me know what approach do you think is best here and how we should go about updating the IB of this ticket.

Thank you.

Reply:
just an idea

If you define `force_active` as a public static class variable you could get the `force_active` value without needing a class instance. That said, I'm not a huge fan of using them because they act like global variables, albeit namespaced global variables. There is likely a cleaner way to do it

Reply:
ugh... I forgot about it :smile:. Yes, let's go with the static approach in this case. But let's use static methods instead of variables. We will need to do the following:

1. Define a new public static method `is_force_active()` in the `Module` class that returns just `false` by default.
2. Update the `Module::parse_info` method to use `static::is_force_active()` method to set the value for the `force_active` property instead of hardcoded `false`: https://github.com/google/site-kit-wp/blob/e6b6890a69213a970cec7c1ca1837e72df9f1150/includes/Core/Modules/Module.php#L690
3. Add `is_force_active()` methods that return `true` to `Search_Console` and `Site_Verification` classes and update the appropriate `setup_info` methods in those classes to use the new static method.
4. Update `Modules::setup_registry` method to check whether the module is active using the new static method.

cc @felixarntz to chime in if he has a better idea.

Reply:
good idea to use methods instead of properties @eugene-manuilov